fgpyo/util/metric.py

```diff
@@ -148,18 +148,25 @@ def _parsers(cls) -> Dict[type, Callable[[str], Any]]:
         return {}
 
     @classmethod
-    def read(cls, path: Path) -> Iterator[Any]:
+    def read(cls, path: Path, skip_extra: bool = True) -> Iterator[Any]:
         """Reads in zero or more metrics from the given path.
 
         The metric file must contain a matching header.
 
         Args:
             path: the path to the metrics file.
+            skip_extra: True to ignore any extra columns, False to raise an exception.
         """
         parsers = cls._parsers()
         with path.open("r") as reader:
             header: List[str] = reader.readline().rstrip("\r\n").split("\t")
-            assert header == cls.header(), "Header did not match"
+            cls_header = cls.header()
+            # check the header
+            for field in cls_header:
+                assert field in header, f"Missing field '{field}' in file: {path}"
+            if not skip_extra:
+                for field in header:
+                    assert field in cls_header, f"Extra field '{field}' in file: {path}"
```

Comment:
For both of these it would be nicer to do:

```python
class_fields = set(cls_header)
file_fields = set(header)
missing_from_class = file_fields.difference(class_fields)
missing_from_file = class_fields.difference(file_fields)

if len(missing_from_class) > 0:
   raise ValueError("Fields in file missing from class: {missing_from_class}")
...
```

Reply:
Also would be helpful to state the class name in the errors along with the file name.